### ISSUE TITLE
r738, TPF Tools, Copy to Clipboard

### ISSUE THREAD

**Issue by ghost** [ME3Explorer/ME3Explorer#208]
@KFreon -- FYI, this doesn't work. Clicking and then opening Notepad doesn't result in being able to copy and paste anything. Not via Ctrl+V or via the Edit > Paste menu. Verified by Creeper, as well.

**Comment by KFreon** [comment]
Ohhhh I remember doing this ages ago.
The idea was that it only worked on selected entries in the PCC panel. Why I didn't name it as such I don't know.
I've changed it so there's two buttons. One for copying selected only, and one for copying all.

**Comment by ghost** [comment]
@KFreon -- Copy All seems to work and copies the PCC list to the clipboard to past via Ctrl+V.

Copy Selected doesn't work.

**Comment by KFreon** [comment]
Copy Selected requires the entries to be selected, not checked?
I say that because it works for me.

**Comment by ghost** [comment]
Doh, okay, let me try again.

**Comment by ghost** [comment]
Okay, that does work. But I think we need a change in language or procedure, @KFreon .

In the same row of buttons we have "Select All", which _means_ "Check All". But then we have "Copy Selected To", which means just that. This isn't intuitive, which is why I missed it.

So, here's what would make most sense to me. Unlike the texture list on the left, the PCC list does _not_ need to have different mechanics for "select" (highlight blue), and "check" (tick). So, why not make them the same? Clicking on the box or the PCC should tick the box and that means "select". Clicking it again unselects it.

This way, the language on all the buttons under the PCC List tab make sense. And there is no difference between select/check, since there doesn't need to be one.

**Comment by KFreon** [comment]
I saw it the other way.
The PCC's are checked by default, but if you only want to copy a subset of them out, you can.
That panel can have things selected with Ctrl and Shift. The checkboxes don't indicate selection like they do on the left, but which PCC's are going to be affected by changes.

So if I did have toggle on selection, they'd all start selected, then you'd select a subset, which would uncheck them. I feel like that isn't what you'd want.

**Comment by ghost** [comment]
@KFreon -- No, I definitely wouldn't want that. What I'm saying is that that _checkmark itself_ should count as what is "selected" and is copied to create the PCC list. I'm saying that the concept of selecting/highlighting is totally unnecessary for the PCC List tab. All we need are the checkmarks; their presence/absence controls what's copied into the PCC list.

Why? Follow my thought train...

If I'm going to export a PCC list for my records, what I want to know are one or two things: 
- all PCCs "containing" the texture
- which PCCs I mod with the new texture

The first scenario is accommodated simply by the PCC appearing in the list, period. The second scenario corresponds exactly to which PCCs are checked.

So, this is how it would work:
1. I load a TPF.
2. I select (highlight) a thumb on the left and then the PCC tab on the right.
3. All PCCs for a texture are checked by default.
4. I uncheck two PCCs that I don't want affected by the new texture. 
5. Now, I want to save a record of the checked PCCs so I know which ones will have the new textures.
6. I click "Copy Selected" to copy my PCC selections. This "reads" which PCCs are checked and copies them to a TXT file/clipboard.

If we want to be very clear, we can change the language on the buttons so everything that says "Checked" instead of "Selected", but I don't think it's necessary. The entire reason I submitted this issue is most people consider those terms interchangeable and the difference isn't intuitive.

The scenario I outline above seems like the most logical way to this feature. Saving a list of either all PCCs OR specifically those that will contain the new texture. In both cases, we only care about what's checked. Having a difference between "checked" and "selected" just confuses things. 

It's entirely possible I'm missing something, though. Are you seeing some other use for "Copy Selected"?

**Comment by KFreon** [comment]
The one I can think of is exporting the pccs you don't have checked, or any subset you don't have checked.

As I recall @CreeperLava was the one who requested this feature. Perhaps he can shed light on the intended purpose.

**Comment by ghost** [comment]
@KFreon -- The problem that I see with that, is exporting/copying a list of what you _don't_ have checked is essentially useless. That information doesn't mean anything unless viewed in the context of which PCCs _could_ be checked. (All PCCs that reference that texture.)

Say I have texture X. It's referenced by 12 PCCs. I only install it to 4. I then dump a list of the 8 PCCs that I didn't install it to. Later on, I look at that list of 8 PCCs. What does it tell me? Not a thing. Sure, it tells me that those 8 PCCs could have the new texture and don't... but which files _do_ have the new texture? That we don't know. To know that, I then need to look up the texture again in Texplorer/TPF Tools and then see which aren't on my list. Or, I _also_ need to dump a list of all 12 PCCs. At which point, you might as well dump the list of 4 in the first place.

Maybe I'm still missing something, though. Let's see what @Creeperlava has to say. If we do keep the separate check/select mechanism that's currently in place, then we need to change the language on the buttons so it's clearer that checked !=selected. More often then not in program usage, these two things are interchangeable, so if it isn't clearer, I see folks getting confused like I was and assuming the feature doesn't work.

**Comment by starkgate** [comment]
I haven't requested this feature @KFreon, you must be thinking of someone else^^. I don't use it either, to be honest. I do agree with giftfish though, checkboxes are the way to go here. Copy the selected PCCs has to apply only to all checked PCCs to avoid any confusion.

**Comment by KFreon** [comment]
Very well. By popular vote, checkboxes it is.

**Comment by ghost** [comment]
Verified in rev748.
